aten/src/ATen/native/Sorting.cpp

```diff
@@ -94,6 +91,169 @@ void quick_select_template(
   } while (true);
 }
 
+// Returns pointer to pivot used to partition range [first, last]
+// There should be at least 3 elements in [first, last]
+template <typename random_it, typename comp>
+random_it partition(random_it first, random_it last, comp lt) {
+  // Use median of three for pivot choice
+  random_it pivot = first + (std::distance(first, last) >> 1);
```

Comment:
To improve readability probably just want to divide by 2. The compiler should figure out that it can do a shift (or not).